### PR TITLE
fix(website): offer to switch to right safe chain

### PR DESCRIPTION
**Pull request by Ignusmart**
The main purpose of this PR is to offer the user to switch to the right chain before executing/staging any transaction

https://github.com/usecannon/cannon/assets/3952453/eaf6339a-a36a-4ca7-b512-492498e2b1ba

### CODE REVIEW

**Review by dbeal-eth — CHANGES_REQUESTED** [review]
in the video you shared, the button appears to un-disable after the switch chain is completed? Shouldnt it continue to be disabled/loading until its done?

also, can we turn this more or less repeated section of code into a helper function? I have a feeling we are going to be using similar pattern a lot.

File: packages/website/src/features/Deploy/QueueDrawer.tsx [hunk 1]
```diff
@@ -380,12 +380,15 @@ export const QueuedTxns = ({
                           colorScheme="teal"
                           w="100%"
                           isDisabled={
+                            stager.signing ||
```
Comment (dbeal-eth) [inline packages/website/src/features/Deploy/QueueDrawer.tsx:383]: in accordance with showing a reason any time a button is disabled, can we put some indication somewhere... somehow on the page that says "currently signing txn" to indicate the button is disabled (maybe the user doenst realize their signing window was opened when they click on the button)

**Review by dbeal-eth — APPROVED** [review]
i still have an umber of things that I think could be improved about this, but we have to set this PRs stopping point somewhere.

File: packages/website/src/features/Deploy/QueueDrawer.tsx (hunk 1 above)
Comment (Ignusmart) [reply packages/website/src/features/Deploy/QueueDrawer.tsx:383]: Nice catch, sure, I'll add it ASAP, thank you

**Review by dbeal-eth — APPROVED** [review]

**Pull request merged**